scripts/build.py

```diff
@@ -1181,32 +1181,42 @@ def generate_build_directory():
     print 'Build completed.'
 
 
-def compile_typescript_files(project_dir):
+def compile_typescript_files(project_dir, compiled_js_dir):
     """Compiles typescript files to produce javascript files in
     local_compiled_js folder.
 
     Args:
         project_dir: str. The project directory which contains the ts files
             to be compiled.
+        compiled_js_dir: str. The directory which contains the compiled js
+            files.
     """
+    safe_delete_directory_tree(compiled_js_dir)
     print 'Compiling ts files...'
     cmd = ['./node_modules/typescript/bin/tsc', '--project', project_dir]
     subprocess.check_call(cmd)
 
 
-def compile_typescript_files_continuously(project_dir):
+def compile_typescript_files_continuously(
```

Comment:
I'm not sure what "continuously" means here, btw -- we seem to return once we see "Watching for file changes". Doesn't that mean this process terminates and is therefore not continuous by definition?

Or did you mean to say "when typescript is done with the initial compilation" in your last comment on lines 1234?

Reply:
Yes, I mean to say that when typescript finishes initial compilation we can return from this function since the watcher is running in the background and the initial set of compiled files is also produced.